billchop-fe/src/pages/GroupSubPage.tsx

```diff
@@ -71,6 +74,33 @@ export default class GroupSubPage extends React.Component<
       });
   };
 
+  getRecentGroupSpendingDatasets = (): IBarChartDataset<number>[] => {
+    let { bills } = this.state;
+    const datasets: IBarChartDataset<number>[] = [];
+    const currentMonth = new Date().getMonth();
+
+    bills = bills.filter((bill) => {
+      const billMonth = new Date(bill.CreationTime).getMonth();
+      return (
+        billMonth === currentMonth ||
+        billMonth === currentMonth - 1 ||
+        billMonth === currentMonth - 2
+      );
+    });
+
+    let month = -1;
+    bills.forEach((bill) => {
+      const billMonth = new Date(bill.CreationTime).getMonth();
+      if (billMonth !== month) {
+        month = billMonth;
+        datasets.push({ label: getMonthName(month), data: 0 });
+      }
+      datasets[datasets.length - 1].data += bill.Total;
+    });
+
```

Comment:
Its most likely going to be cleaner to make datasets a dictionary. The graph lib you are using probably needs an array, but you can get an array out of a dictionary by using ```Object.values(datasets)```. In this case, the code could look something like this:
```typescript
// Need to change dictionary to allow defining key type
const datasets: Dictionary<number, IBarChartDataset<number>>;
// ...
bills.forEach((bill) => {
  const billMonth = new Date(bill.CreationTime).getMonth();
  
  datasets[billMonth] = datasets[billMonth] ?? { label: getMonthName(billMonth), data: 0 };
  datasets[billMonth].data += bill.Total;
});

return Object.values(datasets);
```